extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts

```diff
@@ -0,0 +1,167 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as constants from '../common/constants';
+import * as vscode from 'vscode';
+import * as mssql from 'mssql';
+import * as utils from '../common/utils';
+import type * as azdataType from 'azdata';
+import { PublishDatabaseDialog } from './publishDatabaseDialog';
+import { DeployOptionsModel } from '../models/options/deployOptionsModel';
+
+export class PublishOptionsDialog {
+
+	public dialog!: azdataType.window.Dialog;
+	private optionsTab: azdataType.window.DialogTab | undefined;
+	private disposableListeners: vscode.Disposable[] = [];
+	private descriptionHeading!: azdataType.TableComponent;
```

Comment:
if these variables aren't set in the constructor, the type should have `| undefined`, not a !